eng/pipelines/stress-cluster-provision.yml

```diff
@@ -0,0 +1,20 @@
+pr: none
+
+trigger: none
+
+parameters:
+  - name: Environment
+    type: string
+    default: test
+    values:
+    - prod
+    - test
+  - name: WhatIf
+    type: boolean
```

Comment:
This will be an interesting pattern. If you find that it works out OK can you please write up some guideline docs about this pattern.

Reply:
Yep can do. Plumbing WhatIf support through all our scripts will be pretty useful for general testing and development, I think. Similarly we could run a lot more validation pipelines/scripts on PR changes if we had WhatIf enabled to catch surface level issues.